core/src/main/kotlin/riven/core/service/workflow/engine/coordinator/WorkflowCoordinationService.kt

```
@@ -0,0 +1,397 @@
+            )
+            context.dataRegistry[nodeName] = executionData
+
+            logger.debug { "Captured output for node $nodeName: ${output.keys}" }
+            logger.info { "Data registry now contains ${context.dataRegistry.size} node outputs" }
+
+            NodeExecutionResult(
+                nodeId = nodeId,
+                status = WorkflowStatus.COMPLETED,
+                output = output
+            )
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)